Adjust ADIOS v2.1.1 defaults

Signed-off-by: Mario Roy <...>

diff -uarp a/block/adios.c b/block/adios.c
--- a/block/adios.c
+++ b/block/adios.c
@@ -34,9 +34,9 @@ enum adios_op_type {
 };
 
 // Global variable to control the latency
-static u64 default_global_latency_window = 32000000ULL;
+static u64 default_global_latency_window = 37000000ULL;
 // Ratio below which batch queues should be refilled
-static u8  default_bq_refill_below_ratio = 25;
+static u8  default_bq_refill_below_ratio = 23;
 
 // Dynamic thresholds for shrinkage
 static u32 default_lm_shrink_at_kreqs  =  5000;
-- 
2.40.2

