BORE reweight_entity update from v6.0.0

diff -uarp a/include/linux/sched/bore.h b/include/linux/sched/bore.h
--- a/include/linux/sched/bore.h
+++ b/include/linux/sched/bore.h
@@ -34,7 +34,7 @@ extern void sched_clone_bore(
 extern void reset_task_bore(struct task_struct *p);
 extern void sched_bore_init(void);
 
-extern void reweight_entity(
-	struct cfs_rq *cfs_rq, struct sched_entity *se, unsigned long weight);
+extern void reweight_entity(struct cfs_rq *cfs_rq,
+	struct sched_entity *se, unsigned long weight, bool no_update_curr);
 #endif // CONFIG_SCHED_BORE
 #endif // _LINUX_SCHED_BORE_H
diff -uarp a/kernel/sched/bore.c b/kernel/sched/bore.c
--- a/kernel/sched/bore.c
+++ b/kernel/sched/bore.c
@@ -53,7 +53,7 @@ static void reweight_task_by_prio(struct
 	struct sched_entity *se = &p->se;
 	unsigned long weight = scale_load(sched_prio_to_weight[prio]);
 
-	reweight_entity(cfs_rq_of(se), se, weight);
+	reweight_entity(cfs_rq_of(se), se, weight, true);
 	se->load.inv_weight = sched_prio_to_wmult[prio];
 }
 
diff -uarp a/kernel/sched/fair.c b/kernel/sched/fair.c
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -3933,14 +3933,22 @@ static void reweight_eevdf(struct sched_
 	se->deadline = avruntime + vslice;
 }
 
+#ifdef CONFIG_SCHED_BORE
+void reweight_entity(struct cfs_rq *cfs_rq, struct sched_entity *se,
+			    unsigned long weight, bool no_update_curr)
+#else // !CONFIG_SCHED_BORE
 void reweight_entity(struct cfs_rq *cfs_rq, struct sched_entity *se,
 			    unsigned long weight)
+#endif // CONFIG_SCHED_BORE
 {
 	bool curr = cfs_rq->curr == se;
 	u64 avruntime;
 
 	if (se->on_rq) {
 		/* commit outstanding execution time */
+#ifdef CONFIG_SCHED_BORE
+		if (!no_update_curr)
+#endif // CONFIG_SCHED_BORE
 		update_curr(cfs_rq);
 		avruntime = avg_vruntime(cfs_rq);
 		if (!curr)
@@ -3993,7 +4001,11 @@ static void reweight_task_fair(struct rq
 	struct cfs_rq *cfs_rq = cfs_rq_of(se);
 	struct load_weight *load = &se->load;
 
+#ifdef CONFIG_SCHED_BORE
+	reweight_entity(cfs_rq, se, lw->weight, false);
+#else // !CONFIG_SCHED_BORE
 	reweight_entity(cfs_rq, se, lw->weight);
+#endif // CONFIG_SCHED_BORE
 	load->inv_weight = lw->inv_weight;
 }
 
@@ -4134,7 +4146,11 @@ static void update_cfs_group(struct sche
 	shares = calc_group_shares(gcfs_rq);
 #endif
 	if (unlikely(se->load.weight != shares))
+#ifdef CONFIG_SCHED_BORE
+		reweight_entity(cfs_rq_of(se), se, shares, false);
+#else // !CONFIG_SCHED_BORE
 		reweight_entity(cfs_rq_of(se), se, shares);
+#endif // CONFIG_SCHED_BORE
 }
 
 #else /* CONFIG_FAIR_GROUP_SCHED */
-- 
2.40.2

