
Suppress the variable 'cpu' is uninitialized warnings.

Signed-off-by: Mario Roy <>


diff -uarp a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -1592,7 +1592,10 @@ struct rq *__preempt_rq_trylock(const cp
 			struct rq *rq;
 			cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 			__TRY_LOCK_BEST_RQ(&dst, cpu);
+#pragma clang diagnostic pop
 		}
 	}
 
@@ -1610,7 +1613,10 @@ struct rq *preempt_rq_trylock(const cpum
 			struct rq *rq;
 			cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 			TRY_LOCK_BEST_RQ(&dst, cpu);
+#pragma clang diagnostic pop
 		}
 	}
 
diff -uarp a/kernel/sched/alt_topology.c b/kernel/sched/alt_topology.c
--- a/kernel/sched/alt_topology.c
+++ b/kernel/sched/alt_topology.c
@@ -172,6 +172,8 @@ struct rq *__idle_rq_trylock_func_pcore_
 	cpumask_t dst;
 	cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 	if (cpumask_and(&dst, pmask, sched_idle_mask + 1))
 		__TRY_LOCK_BEST_RQ(&dst, cpu);
 
@@ -180,6 +182,7 @@ struct rq *__idle_rq_trylock_func_pcore_
 
 	if (cpumask_and(&dst, pmask, sched_idle_mask))
 		__TRY_LOCK_BEST_RQ(&dst, cpu);
+#pragma clang diagnostic pop
 
 	return NULL;
 }
@@ -192,6 +195,8 @@ struct rq *idle_rq_trylock_func_pcore_ec
 	cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 	int best_cpu = nr_cpu_ids;
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 	if (cpumask_and(&dst, pmask, sched_idle_mask + 1))
 		TRY_LOCK_BEST_RQ(&dst, cpu);
 
@@ -200,6 +205,7 @@ struct rq *idle_rq_trylock_func_pcore_ec
 
 	if (nr_cpu_ids == best_cpu && cpumask_and(&dst, pmask, sched_idle_mask))
 		TRY_LOCK_BEST_RQ(&dst, cpu);
+#pragma clang diagnostic pop
 
 	*best = best_cpu;
 	return NULL;
@@ -213,11 +219,14 @@ struct rq *__idle_rq_trylock_func_smt(co
 	cpumask_t dst;
 	cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 	if (cpumask_and(&dst, pmask, sched_idle_mask + 1))
 		__TRY_LOCK_BEST_RQ(&dst, cpu);
 
 	if (cpumask_and(&dst, pmask, sched_idle_mask))
 		__TRY_LOCK_BEST_RQ(&dst, cpu);
+#pragma clang diagnostic pop
 
 	return NULL;
 }
@@ -230,11 +239,14 @@ struct rq *idle_rq_trylock_func_smt(cons
 	cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 	int best_cpu = nr_cpu_ids;
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 	if (cpumask_and(&dst, pmask, sched_idle_mask + 1))
 		TRY_LOCK_BEST_RQ(&dst, cpu);
 
 	if (nr_cpu_ids == best_cpu && cpumask_and(&dst, pmask, sched_idle_mask))
 		TRY_LOCK_BEST_RQ(&dst, cpu);
+#pragma clang diagnostic pop
 
 	*best = best_cpu;
 	return NULL;
@@ -245,12 +257,15 @@ struct rq *__idle_rq_trylock_func_defaul
 {
 	cpumask_t dst;
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 	if (cpumask_and(&dst, pmask, sched_idle_mask)) {
 		struct rq *rq;
 		cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 
 		__TRY_LOCK_BEST_RQ(&dst, cpu);
 	}
+#pragma clang diagnostic pop
 
 	return NULL;
 }
@@ -260,12 +275,15 @@ struct rq *idle_rq_trylock_func_default(
 	cpumask_t dst;
 	int best_cpu = nr_cpu_ids;
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
 	if (cpumask_and(&dst, pmask, sched_idle_mask)) {
 		struct rq *rq;
 		cpumask_t *topo_end_mask = per_cpu(sched_cpu_topo_end_mask, cpu);
 
 		TRY_LOCK_BEST_RQ(&dst, cpu);
 	}
+#pragma clang diagnostic pop
 
 	*best = best_cpu;
 	return NULL;
-- 
2.52.0

